src/framework/components/element/text-element.js

```diff
@@ -902,7 +904,8 @@ class TextElement {
 
                 var isWordBoundary = WORD_BOUNDARY_CHAR.test(char);
                 var isCJK = CJK_CHAR.test(char);
-                if (isWordBoundary || isCJK) { // char is space, tab, or dash OR CJK
+                var isNextCJK = (nextchar !== null) ? CJK_CHAR.test(nextchar) : false;
```

Comment:
```suggestion
                const isNextCJK = (nextchar !== null) && CJK_CHAR.test(nextchar);
```

and make the previous two variables const too!